sycl/source/detail/platform_impl.cpp

```diff
@@ -20,6 +20,20 @@ __SYCL_INLINE_NAMESPACE(cl) {
 namespace sycl {
 namespace detail {
 
+static bool IsBannedPlatform(platform Platform) {
+  auto IsNVIDIAOpenCL = [](platform Platform) {
+    if (Platform.is_host())
+      return false;
+
+    const bool HasCUDA = Platform.get_info<info::platform::name>().find(
+                             "NVIDIA CUDA") != std::string::npos;
```

Comment:
This is looking for the NVIDIA OpenCL platform, not the PI CUDA backend, so needs to find a platform with NVIDIA CUDA string (not BACKEND, which would be the PI CUDA backend).
Having the "NVIDIA CUDA" and the `backend::opencl` type means it can only be the NVIDIA OpenCL platform.